File: samples/kmp/shared/src/commonMain/kotlin/jp/kamusoft/ksdialogs/samples/kmp/LayoutDialogViewModel.kt
package jp.kamusoft.ksdialogs.samples.kmp

import jp.kamusoft.ksdialogs.kmp.DialogViewModel

/**
 * Layout Dialog の ViewModel。
 *
 * 結果型は [Boolean] で、完了操作は `true` を報告する。
 * 共有コードで定義したこのクラスがそのまま各 OS のレジストリの登録キーになる。
 *
 * @property message ダイアログに表示するメッセージ
 * @property usesVisibleArea サイズと位置の計算に可視領域を使うか。
 *   静的メタ属性は各 OS の View 定義側で完結するため、View factory が作った View への添付として供給する
 */
class LayoutDialogViewModel(
    val message: String,
    val usesVisibleArea: Boolean,
) : DialogViewModel<Boolean>

File: samples/ios/KsDialogsSample/LayoutDialogViewModel.swift
import KsDialogs

/// Layout Dialog の ViewModel。
///
/// 結果型は `Bool` で、完了操作は `true` を報告する。
/// この型そのものがレジストリの登録キーになる。
final class LayoutDialogViewModel: DialogViewModel {
    typealias Result = Bool

    /// ダイアログに表示するメッセージ。
    let message: String

    /// サイズと位置の計算に可視領域を使うか。
    /// 静的メタ属性は中身の性質なので、View factory が作った View への添付として供給する。
    let usesVisibleArea: Bool

    init(message: String, usesVisibleArea: Bool) {
        self.message = message
        self.usesVisibleArea = usesVisibleArea
    }
}
